imagebuildah/build.go

```diff
@@ -646,10 +664,18 @@ func (b *Executor) Prepare(ctx context.Context, ib *imagebuilder.Builder, node *
 		DefaultMountsFilePath: b.defaultMountsFilePath,
 		Format:                b.outputFormat,
 	}
-	builder, err := buildah.NewBuilder(ctx, b.store, builderOptions)
-	if err != nil {
-		return errors.Wrapf(err, "error creating build container")
+
+	var builder *buildah.Builder
+	var err error
+	if asImageBuilderFound, ok := builderMap[from]; ok {
+		builder = asImageBuilderFound
```

Comment:
This needs to be changed to a fresh container based on an image that's based on the container.  One thing we see done in Dockerfiles is multiple stages referring to the same previous stage, as in https://gist.github.com/bparees/f1c8b734397481af465452bd60ad718e, and in those cases we want changes made in one stage to not affect what the other starts with.

Reply:
This is exactly why I wanted you to review this.  TYVM for the catch and I'll see what I can do to change the code to make this happen.